### PR TITLE
Added wrong env support + list envs

### PR DESCRIPTION
**Pull request by Shivraj0**
[JIRA](https://gofynd.atlassian.net/browse/FPCO-1920)

### CODE REVIEW

File: src/commands/env.ts [hunk 1]
```diff
@@ -2,15 +2,21 @@ import { CommanderStatic } from 'commander';
...
 
 export default function env(program: CommanderStatic) {
-  // List available config
+  
+  // List available environments
   program
-    .command('env')
-    .alias('env ls')
-    .requiredOption('-n, --name [env-name]', 'Environment name')
+    .command('env-ls')
```
Comment (vivek8690) [inline src/commands/env.ts]: a command should be` fdk env ls ` @Shivraj0

File: src/lib/Env.ts [hunk 2]
```diff
@@ -35,15 +36,25 @@ export default class Env {
...
+      
+      Logger.info(chalk.bold.blueBright(`List of supported Environments:`));
+      Object.keys(AVAILABLE_ENVS).forEach(key => {
+        if(key.toString() === ACTIVE_ENVIRONMENT.toString()) {
+          Logger.info(`${chalk.bold.greenBright(key)}*`);
+        } else {
+          Logger.info(chalk.bold.gray(key));
+        }
+      });
+      // if(options.name) {
```
Comment (vivek8690) [inline src/lib/Env.ts]: remove comments @Shivraj0

File: src/lib/Env.ts [hunk 3]
```diff
@@ -33,17 +34,26 @@ export default class Env {
   public static getEnv() {
     const ctx = Env.getEnvValue();
     if(!ctx) {
-      throw new CommandError(COMMON_LOG_MESSAGES.EnvNotSet)
+      throw new CommandError(COMMON_LOG_MESSAGES.EnvNotSet);
     }
-    Logger.success(`Active Envoirnment: ${ctx}`)
+    Logger.success(`${'Active Environment'}: ${chalk.bold(ctx)}`);
```
Comment (vivek8690) [inline src/lib/Env.ts]: `${'Active Environment'}` not needed `Active Environment` will also work

File: src/fdk.ts [hunk 4]
```diff
@@ -14,10 +14,15 @@ import { initializeLogger } from './lib/Logger';
...
 import path from 'path';
+import Env from './lib/Env';
+import { getActiveContext } from './helper/utils.js';
+
 const packageJSON = require('../package.json');
 
 const notRequireAuthCommands = ['login', 'env', 'logout', 'auth'];
-const notRequireEnvCommands = ['env'];
+const notRequireEnvCommands = ['env', 'env-ls'];
+const notRequiredThemeCommands = ['serve', 'sync'];
```
Comment (vivek8690) [inline src/fdk.ts]: This check will be required for all theme-related commands @Shivraj0

Comment (vivek8690) [reply src/fdk.ts]: change variable name too

Comment (Shivraj0) [reply src/fdk.ts]: Done

File: src/lib/Env.ts (hunk 2 above)
Comment (Shivraj0) [reply src/lib/Env.ts]: Done

File: src/commands/env.ts (hunk 1 above)
Comment (Shivraj0) [reply src/commands/env.ts]: Done

File: src/fdk.ts [hunk 5]
```diff
@@ -14,10 +14,15 @@ import { initializeLogger } from './lib/Logger';
...
 import inquirer from 'inquirer';
 import path from 'path';
+import Env from './lib/Env';
+import { getActiveContext } from './helper/utils.js';
+
 const packageJSON = require('../package.json');
 
 const notRequireAuthCommands = ['login', 'env', 'logout', 'auth'];
 const notRequireEnvCommands = ['env'];
+const requiredThemeCommands = ['serve', 'sync', 'new', 'init'];
```
Comment (vivek8690) [inline src/fdk.ts]: define all theme commands as an array in constant and populate use it here @Shivraj0

File: src/lib/Env.ts (hunk 3 above)
Comment (Shivraj0) [reply src/lib/Env.ts]: Done

File: src/fdk.ts (hunk 5 above)
Comment (Shivraj0) [reply src/fdk.ts]: Done @vivek8690

File: src/helper/constants.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const ALL_THEME_COMMANDS = ['new', 'init', 'sync', 'serve', 'publish', 'unpublish', 'pull', 'pull-config', 'context', 'context-list']
+export const THEME_COMMANDS = ['serve', 'sync', 'new', 'init'];
```
Comment (vivek8690) [inline src/helper/constants.ts]: for publish, unpublish, pull, pull-config commands you need to add check for environment mismatch @Shivraj0

Comment (Shivraj0) [reply src/helper/constants.ts]: publish and unpublish will require context and environment check. Rest two i.e pull, pull-config won't require check for the same. @vivek8690

Comment (vivek8690) [reply src/helper/constants.ts]: It will be required for pull and pull-config as well @Shivraj0

Comment (Shivraj0) [reply src/helper/constants.ts]: Done @vivek8690

**Pull request merged**